src/Microsoft.AspNetCore.Sockets.Http/SocketRouteBuilder.cs

```diff
@@ -30,7 +32,20 @@ public void MapSocket(string path, HttpSocketOptions options, Action<ISocketBuil
 
         public void MapEndPoint<TEndPoint>(string path) where TEndPoint : EndPoint
         {
-            MapSocket(path, builder =>
+            MapEndPoint<TEndPoint>(path, socketOptions: null);
+        }
+
+        public void MapEndPoint<TEndPoint>(string path, Action<HttpSocketOptions> socketOptions) where TEndPoint : EndPoint
+        {
+            var authorizeAttributes = typeof(TEndPoint).GetCustomAttributes<AuthorizeAttribute>(inherit: true);
```

Comment:
What? `socketOptions` is just a way for the user to add there own options to the one we make, so we still add our attribute info